hawkbit-core/src/main/java/org/eclipse/hawkbit/exception/EventHandlerChain.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.exception;
+
+/**
+ * A chain where an event is passed through all handlers until one is found that
+ * can process the event.
+ *
+ * @param <T>
+ *            generic type of event.
+ */
+public interface EventHandlerChain<T> {
+
+    /**
+     * Pass the event and search for a suitable handler
+     *
+     * @param event
+     *            the event
+     */
+    void doHandle(final T event);
```

Comment:
why not just `handle`? I would expect that `doHandle` to be called from a `handle` function. Something like this:
```java
public void handle(T error) {
   if(error instanceof Throwable) {
      doHandle((Throwable) error);
      return;
   } else {
      // do something else
   }
}

public void doHandle(Throwable error) {
   // todo
}
```
So I wouldn't expect a `doHandle` function to exist before having a `handle` function in place

Reply:
I think in the case of `Filter::doFilter` which is our inspiration for the handler chain here, they maybe name the function like that because it would otherwise be weird to name it `filter` : `Filter::filter`

Reply:
👍